src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -735,5 +737,24 @@ private async Task UpdateSingleInstallerInteractively(Installer installer)
             Console.ReadKey();
             Console.Clear();
         }
+
+        private string GetPRTitle(Manifests repositoryManifest)
+        {
+            // Use custom PR title if provided by the user.
+            if (!string.IsNullOrEmpty(this.PRTitle))
+            {
+                return this.PRTitle;
+            }
+
+            // Handle case where manifest is still using singleton format
+            string repositoryVersion = repositoryManifest.SingletonManifest != null ? repositoryManifest.SingletonManifest.PackageVersion : repositoryManifest.InstallerManifest.PackageVersion;
+
+            return WinGetUtil.CompareVersions(this.Version, repositoryVersion) switch
+            {
+                > 0 => $"New version: {this.Id} version {this.Version}",
+                < 0 => $"Add version: {this.Id} version {this.Version}",
+                _ => $"Update version: {this.Id} version {this.Version}",
+            };
+        }

```

Comment:
I feel that this would be better suited in the BaseCommand class as a shared helper function for all commands. Does the submit command also need to call this function to generate its own PR title?

Reply:
Refactored the method into BaseCommand Class.

Reply:
Currently, the Submit command can't use this function to get its PR title since submit command can be used for any of the conditions (new ver, update, old ver, new pkg). To use this function, Submit command would need to be modified to fetch the manifest from the repo like how Update Command does. The current behavior is that submit command doesn't give any PRTitle info to SubmitGitHubManifest() and the PRtitle for it is set later in GitHub.cs

https://github.com/microsoft/winget-create/blob/aac1dbf8646da27dbb8793bbccecf65d0e89989d/src/WingetCreateCore/Common/GitHub.cs#L307

I'm not sure if we should be fetching an existing manifest in SubmitCommand just to get its PRTitle but I would like some feedback here...

Reply:
Possible option would be either one of:
1. Fetch existing latest manifest from packages repo in SubmitCommand to be sure of type of PR
2. Use a default value in SubmitCommand and don't use optional parameters in SubmitGitHubManifest() (so that we don't need to set it later in GitHub.cs)
3. Leave it as is

Reply:
Lets go with option 3. I don't want to overcomplicate the submit command by having to fetch existing manifests just to submit a manifest. I think the default PR title in `GitHub.cs` should be kept as a fallback for all submissions so it is okay to have the submit command pass in `null` for the prtitle. Thanks for the thorough explanation.